add warning about uninstall with filename (#7943)
istio/istio.io: content/en/docs/setup/upgrade/index.md

$ istioctl x uninstall --revision istio-1-6-5
{{< /text >}}

{{< warning >}}

> This command will also remove gateway deployments for specified revision. If you want to remove the canary revision, please reinstall the gateway(s) with previous revision first to avoid down time.

Reviewer: Is this correct all the time?  I think it won't remove if I'm deleting the old revision (1-6-5) but would if i'm deleting the new revision (1-7-0).

Reviewer: I propose to have 2 sections, 1 for remove old cp, 1 for remove the new version (canary).  This warning only applied to the 2nd scenario.

Reviewer 2: I added some text for this along with another fix in #7957. If that looks alright, then we can close this PR.

Author: @linsun yes, that's why I emphasized `canary` one here. But if that's not clear enough, then different section is ok

$ istioctl x uninstall -f manifests/profiles/default.yaml
{{< /text >}}

{{< warning >}}

> Note that this command would also remove gateway deployments, use it with caution. The issue would be fixed in release `1.7.1`

Reviewer 2: ```suggestion
This command will also remove gateway deployments, which will then need to be redeployed. The issue will be fixed in release `1.7.1`
```

Reviewer: @richardwxn if the fix is NOT in 1.7.0, can we just recommend users how to clean up manually without uninstall cmd?

Using uninstall cmd would create a non working cluster for users without the fix.

Author: @linsun the use cases of --revision and --purge works, which I expect most people would use, I can remove the part of -f, wdyt?

Reviewer: Looks like the fix will land in 1.7.0 afterall.  If --revision works for default, I agree with removing the -f portion.

Author: updated the description
